examples/spotprice/main.go

```diff
@@ -0,0 +1,55 @@
+/*
+An example tool to find the best provider to launch an
+spot instance based on the specifications provided:
+cpus, memory, and os
+
+Credentials needs to setup as expected by the cloud provider
+SDKs, underneath mapt calls the credential setup helpers
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/redhat-developer/mapt/pkg/spot"
+)
+
+func main() {
+	// The SpotRequest struct holds the desired
+	// specification in terms of hardware specs
+	// and OS requirements
+	spotReq := spot.SpotRequest{
+		CPUs:       4,
+		MemoryGib:  8,
+		Arch:       "amd64",
+		NestedVirt: false,
+		Os:         "fedora",
+		OSVersion:  "39",
+	}
+
+	// Get the lowest price for only aws
+	//spi, err := spotReq.GetAwsLowestPrice()
+
```

Comment:
I guess this is expecting to have the specific Envs with credentials ...can we add that here? I mean os.SetEnv(AWS..., SampleSecretKey)....

To make more clear how we are expecting the credentials to be set?

Reply:
updated with the required env variables that needs to be set for both aws and azure